test/tap/umask.js

```diff
@@ -0,0 +1,64 @@
+var path = require("path")
+var test = require("tap").test
+var rimraf = require("rimraf")
+var mkdirp = require("mkdirp")
+var common = require("../common-tap.js")
+var npmlog = require("npmlog")
```

Comment:
arguably we should be checking that the warnings are printed.  Is there an easier way to do that than monkeypatching npmlog.warn ?